unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -162,31 +268,73 @@ public static int array_length([NativeCallbackType("MonoArray*")] IntPtr array)
         [NativeCallbackType("MonoClass*")] IntPtr klass)
         => FormatterServices.GetUninitializedObject(klass.TypeFromHandleIntPtr()).ToNativeRepresentation();
 
+    [return: NativeCallbackType("MonoException*")]
+    public static IntPtr exception_from_name_msg(
+        [NativeCallbackType("MonoImage*")] IntPtr image,
+        [NativeCallbackType("const char*")] sbyte* name_space,
+        [NativeCallbackType("const char*")] sbyte* name,
+        [NativeCallbackType("const char*")] sbyte* msg)
+    {
+        Type type = class_from_name(image, name_space, name, false).TypeFromHandleIntPtr();
+        return Activator.CreateInstance(type, new string(msg)).ToNativeRepresentation();
+    }
+
     [return: NativeCallbackType("MonoObject*")]
     public static IntPtr type_get_object(
         [ManagedWrapperOptions(ManagedWrapperOptions.Exclude)] [NativeCallbackType("MonoDomain*")]
         IntPtr domain,
         [NativeCallbackType("MonoType*")] IntPtr type)
         => type.TypeFromHandleIntPtr().ToNativeRepresentation();
 
+    public static bool unity_class_has_attribute(
+        [NativeCallbackType("MonoClass*")] IntPtr klass,
+        [NativeCallbackType("MonoClass*")] IntPtr attr_klass)
+    {
+        return Attribute.GetCustomAttribute(klass.TypeFromHandleIntPtr(), attr_klass.TypeFromHandleIntPtr()) != null;
+    }
+
+    public static bool unity_assembly_has_attribute(
+        [NativeCallbackType("MonoAssembly*")] IntPtr assembly,
+        [NativeCallbackType("MonoClass*")] IntPtr attr_klass)
+    {
+        return Attribute.GetCustomAttribute(GetAssemblyForHandle(assembly), attr_klass.TypeFromHandleIntPtr()) != null;
+    }
+
+    public static bool unity_method_has_attribute(
+        [NativeCallbackType("MonoMethod*")] IntPtr method,
+        [NativeCallbackType("MonoClass*")] IntPtr attr_class)
+    {
+        MethodBase mb = MethodBase.GetMethodFromHandle(method.MethodHandleFromHandleIntPtr());
+        return mb != null && mb.GetCustomAttribute(attr_class.TypeFromHandleIntPtr()) != null;
+    }
+
+    [return: NativeCallbackType("const char*")]
+    public static IntPtr image_get_name([NativeCallbackType("MonoImage*")] IntPtr image)
+    {
+        if (m_assemblies.TryGetValue(GetAssemblyForHandle(image), out var pair))
```

Comment:
I think you need to add the image in this case if it's not found, rather than returning `IntPtr.Zero`.